[CI] ClusterPersistentTasksCustomMetadataTests testMinVersionSerialization failing

**Build Scans:**
- [elasticsearch-intake #32426 / part1](https://gradle-enterprise.elastic.co/s/agypcqkzv4usc)
- [elasticsearch-pull-request #110312 / part-1](https://gradle-enterprise.elastic.co/s/l4kyaufraeh6g)
- [elasticsearch-pull-request #110269 / part-1](https://gradle-enterprise.elastic.co/s/fagt36chvi3gq)
- [elasticsearch-intake #32378 / part1](https://gradle-enterprise.elastic.co/s/yiuqsvg56npf6)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.persistent.ClusterPersistentTasksCustomMetadataTests.testMinVersionSerialization" -Dtests.seed=25698C6ECF330181 -Dtests.locale=ee-TG -Dtests.timezone=America/Havana -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.persistent.ClusterPersistentTasksCustomMetadataTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testMinVersionSerialization'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: iterable containing ["test_compatible_version"]
     but: not matched: "test_incompatible_version"
```

**Issue Reasons:**
- [main] 4 failures in test testMinVersionSerialization (1.5% fail rate in 265 executions)
- [main] 2 failures in step part1 (4.1% fail rate in 49 executions)
- [main] 2 failures in step part-1 (1.2% fail rate in 162 executions)
- [main] 2 failures in pipeline elasticsearch-intake (4.1% fail rate in 49 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.2% fail rate in 161 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -1,7 +1,4 @@
 tests:
-- class: "org.elasticsearch.client.RestClientSingleHostIntegTests"
-  issue: "https://github.com/elastic/elasticsearch/issues/102717"
-  method: "testRequestResetAndAbort"
 - class: org.elasticsearch.packaging.test.PackagesSecurityAutoConfigurationTests
   method: test20SecurityNotAutoConfiguredOnReInstallation
   issue: https://github.com/elastic/elasticsearch/issues/112635
@@ -457,12 +454,6 @@ tests:
 - class: org.elasticsearch.index.mapper.LongSyntheticSourceNativeArrayIntegrationTests
   method: testSynthesizeArrayRandom
   issue: https://github.com/elastic/elasticsearch/issues/139562
-- class: org.elasticsearch.persistent.ClusterPersistentTasksCustomMetadataTests
-  method: testMinVersionSerialization
-  issue: https://github.com/elastic/elasticsearch/issues/139624
-- class: org.elasticsearch.persistent.PersistentTasksCustomMetadataTests
-  method: testMinVersionSerialization
-  issue: https://github.com/elastic/elasticsearch/issues/139625
 
 # Examples:
 #

--- a/server/src/test/java/org/elasticsearch/persistent/BasePersistentTasksCustomMetadataTests.java
+++ b/server/src/test/java/org/elasticsearch/persistent/BasePersistentTasksCustomMetadataTests.java
@@ -31,7 +31,6 @@ import java.util.Optional;
 import static org.elasticsearch.cluster.metadata.Metadata.CONTEXT_MODE_GATEWAY;
 import static org.elasticsearch.cluster.metadata.Metadata.CONTEXT_MODE_SNAPSHOT;
 import static org.elasticsearch.persistent.PersistentTasksExecutor.NO_NODE_FOUND;
-import static org.elasticsearch.test.TransportVersionUtils.getFirstVersion;
 import static org.elasticsearch.test.TransportVersionUtils.getNextVersion;
 import static org.elasticsearch.test.TransportVersionUtils.getPreviousVersion;
 import static org.elasticsearch.test.TransportVersionUtils.randomVersionBetween;

--- a/server/src/test/java/org/elasticsearch/persistent/BasePersistentTasksCustomMetadataTests.java
+++ b/server/src/test/java/org/elasticsearch/persistent/BasePersistentTasksCustomMetadataTests.java
@@ -201,7 +201,7 @@ public abstract class BasePersistentTasksCustomMetadataTests<T extends Metadata.
     public void testMinVersionSerialization() throws IOException {
         PersistentTasks.Builder<?> tasks = builder();
 
-        TransportVersion minVersion = getFirstVersion();
+        TransportVersion minVersion = TransportVersion.minimumCompatible();
         TransportVersion streamVersion = randomVersionBetween(random(), minVersion, getPreviousVersion(TransportVersion.current()));
         tasks.addTask(
             "test_compatible_version",

